exec/internal/compile/scanner.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 The go-interpreter Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package compile
+
+import (
+	ops "github.com/go-interpreter/wagon/wasm/operators"
+)
+
+type scanner struct {
+	supportedOpcodes map[byte]bool
+}
+
+// InstructionMetadata describes a bytecode instruction.
+type InstructionMetadata struct {
+	Op    byte
+	Start int
+	Size  int
+}
+
+// CompilationCandidate describes a range of bytecode that can
+// be translated to native code.
+type CompilationCandidate struct {
+	// Bytecode index of the first opcode.
```

Comment:
for short doc strings like that, it looks nicer to put it on the same line than the field being documented.